idaes/core/util/tests/test_model_diagnostics.py

```diff
@@ -975,10 +975,10 @@ def test_display_near_parallel_variables(self):
         model.v3 = Var()
         model.v4 = Var()
 
-        model.c1 = Constraint(expr=model.v1 == model.v2 - 0.99999 * model.v4)
-        model.c2 = Constraint(expr=model.v1 + 1.00001 * model.v4 == 1e-8 * model.v3)
+        model.c1 = Constraint(expr=1e-8 * model.v1 == 1e-8 * model.v2 - 1e-8 * model.v4)
+        model.c2 = Constraint(expr=1e-8 * model.v1 + 1e-8 * model.v4 == model.v3)
         model.c3 = Constraint(
-            expr=1e8 * (model.v1 + model.v4) + 1e10 * model.v2 == 1e-6 * model.v3
+            expr=1e3 * (model.v1 + model.v4) + 1e3 * model.v2 == model.v3
```

Comment:
This test was pretty fragile. The coefficients on `v3` are small relative to those on other variables, so its dot product is vacuously parallel with anything. This didn't show up previously because we were relying on the absolute tolerance of 1e-4 to give it an empty sparsity pattern. Maybe this raises the question of what to do when one of the vectors we are testing is zero. I've removed the coefficients on `v3` so that this isn't an issue for now.

Also, the 1e-8 ratio between small and large coefficients for `v1` and `v2` lead to relative "distances-from-parallel" that are very close to our new tolerance of 1e-8, so I've adjusted some coefficients to give us a little more buffer.

Reply:
This is really because the relative tolerance should be comparing to is `diff <= tolerance * unorm *vnorm`, not `diff <= tolerance * max(unorm, vnorm)`. The absolute tolerance probably can get phased out entirely.

Reply:
Maybe `unorm * vnorm` is the right denominator for the relative tolerance. I would still like to have an absolute tolerance, though, as I wouldn't want `(1e-9, 2e-9)` to be considered parallel to `(1, 2)`. (Unless we want to consider a zero vector as "parallel to everything", which I think is debatable.)

Reply:
I support filtering out rows/columns with norms that are effectively zero, because otherwise they'll just spam the log as being parallel to every other row. No need to use an absolute tolerance for the inner product, though, we can just choose to skip displaying the row/column if one of the norms is too small.

In the diagnostics toolbox, the tolerance for this filtering should be the `CONFIG.jacobian_small_value_warning`, the same tolerance as used in `display_variables_with_extreme_jacobians` and `display_constraints_with_extreme_jacobians` to display a warning message. That way it's displayed (once) to the user there.